gateway/src/resty/http_ng/request.lua

```diff
@@ -20,13 +23,20 @@ local request = { }
 
 request.headers = require 'resty.http_ng.headers'
 
-local function extract_host(url)
-  local _, last = find(url, '://', 0, true)
-  local len = find(url, '/', last + 1, true)
+local function host_header(uri)
+    local port = uri.port
+    local default_port = resty_url.default_port(uri.scheme)
 
-  if len then len = len - 1 end
+    if port and port ~= default_port then
```

Comment:
I've seen in the JIRA issue that this strips the port only if it's the default one to fix some issue when using OIDC. However, this is a generic client, doesn't it affect other use cases?

Reply:
No, not a issue at all.